api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -133,14 +133,13 @@ message HttpConnectionManager {
   // header in responses. If not set, the default is *envoy*.
   string server_name = 10;
 
-  // The maximum request headers size for incoming connections. The default max
-  // is 60K, based on default settings for http codecs. For HTTP1, the current
-  // limit set by http_parser is 80K. for HTTP2, the default allowed header
-  // block in nghttp2 is 64K. The max configurable setting is 64K in order to
-  // stay under both codec limits.
-  // Requests that exceed this size will receive a 431 response.
+  // The maximum request headers size for incoming connections.
+  // If unconfigured, the default max request headers allowed is 60 KiB.
+  // Requests that exceed this limit will receive a 431 response.
+  // The 96 KiB limit is based on bugs in nghttp2 parsing, and can be raised once those are
```

Comment:
For my knowledge, what bugs are we talking about here? Either way, I would probably leave this out of the public docs and just put a comment somewhere in the code on what is going on.

Reply:
Edited the comment as you asked, but I don't know what the bug is exactly. Just, some exception was thrown around 97 or 100kb of headers sent and it wasn't relevant to me to push it further.